adr/keycloak-adr.md

```diff
@@ -1,264 +1,139 @@
+
+**Cons:**
+- More Work: You'll have to write and maintain the code yourself, which can be a lot of work.
+
+### Deprovision Users
+Keycloak will be used to de-provision users and manage user access rights, ensuring timely revocation of access when users leave the organization or change roles.
+
+#### Automated Deprovisioning via SCIM
+**Pros:**
+- Ensures Timely Revocation of Access: Automatically removes users' access rights as soon as their status changes, improving security.
+- Reduces Manual Workload: Lessens the burden on IT staff by handling the de-provisioning process automatically, minimizing the chance of oversight.
+Supports Real-Time Updates: It can integrate with broader system architectures to ensure updates are processed immediately, keeping user access rights current.
+
+**Cons:**
+- Requires Careful Configuration: Setting up SCIM to work correctly with GitLab and Keycloak requires careful planning and configuration.
+- Potential for Misconfiguration: Misconfigurations can lead to unauthorized access or the accidental removal of access rights, impacting operations and security.
+- Complexity: Integrating SCIM with existing systems can be complex.
+
+#### Manual Deprovisioning
+**Pros:**
+Direct Control Over the Process: This gives administrators complete oversight of when and how users are deprovisioned, ensuring no automated system can mistakenly alter access rights.
+- Simple to Implement: It doesn't require introducing new tools or technologies beyond what's already in use, making it straightforward to execute.
+
+**Cons:**
+Labor-Intensive: Each de-provisioning event requires manual action, which can be time-consuming, especially for organizations with a high volume of user changes.
+- Prone to Delays: The manual process is susceptible to delays, which can lead to unauthorized access if users without access retain it due to oversight or backlog.
+- Not Scalable:
+
+#### Custom Code
+**Pros:**
+- Flexibility: You can tailor the de-provisioning process to your exact needs, ensuring it fits your organization perfectly.    
+
+**Cons:**
+- More Work: You'll have to write and maintain the code yourself, which can be a lot of work.
+
+#### GitLab Group Integration
+
+#### Automated Group Management via Keycloak
+**Pros:**
+- Easier Access Management: Linking GitLab with Keycloak groups means when someone's group membership changes in Keycloak, it updates in GitLab too. This keeps access rights tight without extra effort.
+- Automatic Updates: Group memberships are updated automatically, saving a lot of manual work and ensuring no one is accidentally left in or out of a group.
+- Better Security: With automatic updates, you can be sure that everyone has the access they need and nothing more, keeping your projects and data safer.
+
+**Cons:**
+- Requires Careful Configuration: Setting up Keycloak to work correctly with GitLab requires careful planning and configuration.
+- Learning Curve: 
```

Comment:
```suggestion
**Cons:**
- Requires administering groups from Keycloak instead of GitLab. This may be more complex and not all users may have Keycloak group admin access. (However, this may scale better to large numbers of groups.)
- Requires Careful Configuration: Setting up Keycloak to work correctly with GitLab requires careful planning and configuration.
- Learning Curve: more complex to administer groups from Keycloak instead of GitLab. Another admin tool to learn and some complexity from not using GitLab directly.
```